src/shogun/kernel/string/HistogramWordStringKernel.cpp

```diff
@@ -434,23 +434,17 @@ void HistogramWordStringKernel::init()
 	/*m_parameters->add_vector(&variance, &num_params2, "variance");*/
 	watch_param("variance", &variance, &num_params2);
 
-	SG_ADD(&estimate, "estimate", "Plugin Estimate.");
-
-	add_callback_function("estimate", [this](){
-		if (!std::dynamic_pointer_cast<PluginEstimate>(estimate)) {
-			error("Expected Machine to be PluginEstimate");
-			estimate.reset();
-		}
-	});
+	SG_ADD(&estimate, "estimate", "Plugin Estimate.", ParameterProperties::CONSTRAIN,
+		SG_CONSTRAINT(dynamic_cast_checker<std::shared_ptr<Machine>, PluginEstimate>()));
 }
 
 #ifdef DEBUG_HWSK_COMPUTATION
-float64_t CHistogramWordStringKernel::compute_slow(int32_t idx_a, int32_t idx_b)
+float64_t HistogramWordStringKernel::compute_slow(int32_t idx_a, int32_t idx_b)
```

Comment:
should rather delete all this code....debugging with ifdef is something we try to avoid these days I guess? :)

Reply:
sure, I was just fixing the typos :p